implemented custom accent theme selection

Deepanshu-Rohilla opened the pull request:
1. Implemented custom accent theme selection from settings. Showed a palette for various colors.
2. Available theme selection are: `light`, `dark` and `follow system`. Follow system will adapt the mobile theme (dark or light). Note that if we change the theme of mobile while `follow system` is chosen, it would automatically adapt to the change.
3. Unfortunately, #91 is still there. However, effect is almost negligible. It is slightly significant if someone choose two completely different colors. Good thing is that it is gone when someone switch between themes (because color remains same).
This is how the app looks in various themes:


![WhatsApp Video 2020-07-13 at 12 51 33 AM](https://user-images.githubusercontent.com/52520071/87254940-315bc580-c4a4-11ea-9d50-6c45a3bbed90.gif)

kshitijalwadhi commented:
Terrific work. 🤩

A few things that you can look into:
* When the user has requested for a group, the border of that group in tripslist.dart has been hardcoded to `pink[300]`. And the current group is accent colour. See if you can add a condition on that `pink[300]` so that it is some other colour in the cases when the accent colour is similar to pink. For other colours it will work fine.
* See if you can do something to indicate that there are more colours to be selected in that row of accent colours. Maybe add something like a scrollbar? 
* Add a sizedbox after the 'App Theme' tile in settings to make the distance between the tiles same.

Deepanshu-Rohilla commented:
![WhatsApp Image 2020-07-13 at 2 13 34 PM](https://user-images.githubusercontent.com/52520071/87286816-f47ce680-c516-11ea-979d-6ba4128c8291.jpeg)


1. Adjusted the spacing between widgets.
2. Specified color themes wherever required (on sign in, register etc.) 
3. Also fixed Bug Icon position.
